### ISSUE TITLE
Wrong size is being reported by ScaleFactorChanged on Android some times

### ISSUE THREAD

**Issue by limikael** [rust-windowing/winit#2760]
Please see this video for the symptom:

https://www.youtube.com/watch?v=8g2HpXj9V2M

When the phone is in landscape mode, the screen size should be 2307 x 1080, in portrait 1080 x 2400. As you can see from the log in the background, they are correct about 50%-80% of the time, but in the other cases the old values are reported.

**Comment by MarijnS95** [comment]
Known regression when `winit` migrated from `ndk-glue` to `android-activity`. Fix is up at https://github.com/rust-mobile/android-activity/pull/70, will close as this is not a `glutin` bug and the issue+solution is already being tracked elsewhere.

**Comment by kchibisov** [comment]
@MarijnS95 it's a winit repo already.

**Comment by MarijnS95** [comment]
@kchibisov Ah the transfer didn't show up. Regardless, it's an `android-mobile` bug, not a `winit` bug (the `android-mobile` crate simply omitted some things for `native-activity`, once implemented `winit` only needs to update).

**Comment by kchibisov** [comment]
Yeah, it's just better to track things like that, we have issues for dependencies in winit as well.

**Comment by MarijnS95** [comment]
I don't think `Fixes: https://github.com/rust-windowing/winit/issues/2760` in a different _orga_ works, but GitHub now says `linked a pull request [now ](https://github.com/rust-windowing/winit/issues/2760#ref-pullrequest-1636778223)that will close this issue` for the PR in question.

**Comment by kchibisov** [comment]
I mean in general, if issue is reported to the repo and it's an upstream dependency issue causing issue for an application, it's better to track here as well, given that winit needs a dependency upgrade.

If you make a patch bump, then you close after patch bump if you need a breaking change, you do a PR bumping.

Just put `Upstream` label if something is upstream related.

**Comment by MarijnS95** [comment]
> I don't think `Fixes: https://github.com/rust-windowing/winit/issues/2760` in a different _orga_ works, but GitHub now says `linked a pull request now that will close this issue` for the PR in question.

Looks like it worked 🎉
